src/Microsoft.Framework.Runtime/Servicing/ServicingIndex.cs

```diff
@@ -28,54 +28,61 @@ public void Initialize(string folderPath)
             }
             using (var stream = new FileStream(indexFilePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite | FileShare.Delete))
             {
-                using (var reader = new StreamReader(stream))
+                Initialize(servicingFolder, stream);
+            }
+        }
+
+        internal void Initialize(string patchesFolder, Stream indexStream)
+        {
+            _patchesRootFolder = patchesFolder;
+
+            using (var reader = new StreamReader(indexStream))
+            {
+                var lineNumber = 0;
+                for (var line = reader.ReadLine(); line != null; line = reader.ReadLine())
                 {
-                    var lineNumber = 0;
-                    for (var line = reader.ReadLine(); line != null; line = reader.ReadLine())
-                    {
-                        lineNumber += 1;
+                    lineNumber += 1;
 
-                        if (line.StartsWith("#") || string.IsNullOrWhiteSpace(line))
-                        {
-                            continue;
-                        }
-                        var parts = line.Split(new[] { '=' }, 2);
-                        if (parts.Length != 2)
+                    if (line.StartsWith("#") || string.IsNullOrWhiteSpace(line))
+                    {
+                        continue;
+                    }
+                    var parts = line.Split(new[] { '=' }, 2);
+                    if (parts.Length != 2)
+                    {
+                        Logger.TraceInformation("[{0}]. Line ({1}): malformed servicing file", GetType().Name, lineNumber);
+                        continue;
+                    }
+                    parts[0] = parts[0].Trim();
+                    parts[1] = parts[1].Trim();
+                    if (parts[0].Length == 0 || parts[1].Length == 0)
```

Comment:
This code was already there :) I only changed the log message and extracted a new method